repos/fdbt-site/src/pages/api/searchOperators.ts

```diff
@@ -50,44 +34,28 @@ export const getOperatorsWithoutServices = async (selectedOperators: Operator[])
     return results.filter((item) => item !== '');
 };
 
+export const isSearchInputValid = (searchText: string): boolean => {
+    const searchRegex = new RegExp(/^[a-zA-Z0-9\-:\s]+$/g);
+    return searchRegex.test(searchText);
+};
+
 export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
     try {
         const errors: ErrorInfo[] = [];
-        const {
-            search,
-            searchText,
-            addOperators,
-            operatorsToAdd,
-            removeOperators,
-            operatorsToRemove,
-            continueButtonClick,
-        } = req.body;
+        const { search, searchText, userSelectedOperators, continueButtonClick } = req.body;
 
         const multiOperatorAttribute = getSessionAttribute(req, MULTIPLE_OPERATOR_ATTRIBUTE);
         let selectedOperators = multiOperatorAttribute ? multiOperatorAttribute.selectedOperators : [];
 
-        if (removeOperators) {
-            if (!operatorsToRemove) {
-                errors.push({
-                    errorMessage: 'Select at least one operator to remove',
-                    id: removeOperatorsErrorId,
-                });
-            } else {
+        if (search) {
```

Comment:
if search && userSelectedOperators 

would this not be more concise?

Reply:
doing so means we have to write lines (56 - 74) twice, because they should run in both cases. or make it more complicated. for the rest of the search functionality